lib/game.rb

```diff
@@ -0,0 +1,94 @@
+class Game
+	attr_accessor :board, :player_1, :player_2
+	WIN_COMBINATIONS = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]]
+
+
+	def initialize(player_1 = Player::Human.new("X"), player_2 = Player::Human.new("O"), board = Board.new)
+		@board = board
+		@player_1 = player_1
+		@player_2 = player_2
+		
+	end
+
+	def current_player
+		if self.board.turn_count.odd?
+			return self.player_2
+		else
+			return self.player_1
+		end
+	end
+
+	def over?
+		board.full? || self.won? || self.draw?
+	end
+
+	def won?
+			a = []
+			WIN_COMBINATIONS.each do |i|
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.find do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```